android/src/test/java/com/rudderstack/android/internal/infrastructure/LifecycleObserverPluginTest.kt

```diff
@@ -16,5 +91,117 @@ class LifecycleObserverPluginTest {
         lifecycleObserverPlugin.onAppBackgrounded()
 
         verify { analytics.flush() }
+        lifecycleObserverPlugin.shutdown()
+    }
+
+    @Test
+    fun `test when app foregrounded from_background should be true and contain version first time`() {
+        val plugin = LifecycleObserverPlugin()
+        plugin.setup(analytics)
+        plugin.onAppForegrounded()
+        busyWait(100)
+        analytics.assertArgument { input, _ ->
+            assertThat(
+                input, allOf(
+                    isA(TrackMessage::class.java),
+                    hasProperty("eventName", `is`(EVENT_NAME_APPLICATION_OPENED)),
+                    hasProperty(
+                        "properties", allOf<TrackProperties>(
+                            aMapWithSize(2),
+                            hasEntry("from_background", false),
+                            hasEntry("version", "1.0"),
+                        )
+                    )
+                )
+            )
+        }
+        plugin.shutdown()
+
+    }
+
+    @Test
+    fun `test when app foregrounded from_background should be false and not contain version second time`() {
+        val plugin = LifecycleObserverPlugin()
+        plugin.setup(analytics)
+        plugin.onAppForegrounded()
+        busyWait(100)
+        plugin.onAppForegrounded()
+        analytics.assertArgument { input, _ ->
+            assertThat(
+                input, allOf(
+                    isA(TrackMessage::class.java),
+                    hasProperty("eventName", `is`(EVENT_NAME_APPLICATION_OPENED)),
+                    hasProperty(
+                        "properties", allOf<TrackProperties>(
+                            aMapWithSize(1),
+                            hasEntry("from_background", true),
```

Comment:
The test case name `test when app foregrounded from_background should be false` and this `hasEntry("from_background", true)` verification doesn't match. Please correct this.